### PR TITLE
Only switch to NM if not already done

### PR DESCRIPTION
**Pull request by asdil12**
https://progress.opensuse.org/issues/41429
Verification run: http://artemis.suse.de/tests/514#step/yast2_network_setup/2

### CODE REVIEW

File: tests/x11/network/yast2_network_setup.pm [hunk 1]
```diff
@@ -21,7 +21,14 @@ use utils;
 
 sub run {
     my $self = shift;
+
+    select_console 'root-console';
+    my $nm_is_active = (script_run("readlink /etc/systemd/system/network.service | grep NetworkManager") == 0);
```
Comment (rwx788) [inline tests/x11/network/yast2_network_setup.pm]: Could you please extract it to the separate method to `lib/network_utils.pm`? As we could reuse it at least in yast2_lan and yast2_lan_restart.
Also, we could have checked if having NM matches our expectations, as we don't expect it to be on SLE.

File: tests/x11/network/yast2_network_setup.pm [hunk 2]
```diff
@@ -21,7 +21,14 @@ use utils;
```
[the remark this answers is not in the document]
Comment (asdil12) [reply tests/x11/network/yast2_network_setup.pm]: Done.
I also renmed the test module as `yast2_network_setup` wasn't really describing what the module was doing.
The new name is `yast2_network_use_nm`.

**Comment by okurz** [comment]
@rwx788 so PR merged but now for the next step, how to use in the yast modules you mentioned?

**Comment by rwx788** [comment]
@jknphy could you please take a look how we can reuse this instead of that needle detection you've worked on?

**Comment by jknphy** [comment]
I did it at the level of product in https://openqa.suse.de/tests/overview?distri=sle&version=12-SP4&build=0406&groupid=139 and https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/5788/files#diff-d3e0a6c7df772cff7741832a6250d5afR1446. The needle is still needed if want to accept the warning/info pop-up dialog, because it is true that we are not testing almost anything but at least we enter in the module and check the tabs. 
On the other hand, as a second option if we don't even want to schedule those tests, we can delete the code to accept the warning (including the needle) and put outside in a library the function (as you mentioned) to detect it and then unschedule them at the level of main.pm.

**Comment by rwx788** [comment]
@jknphy We cannot use this one in main.pm as we access SUT, I just though that potentially can reuse it, so in case module is scheduled, we check if NM is there and do some NM specific asserts. Alright, let's consider it for future needs. Thanks for the reply!

**Pull request merged**